src/wp-includes/theme.php

```diff
@@ -4352,6 +4352,11 @@ function create_initial_theme_features() {
  * @return bool Whether the active theme is a block-based theme or not.
  */
 function wp_is_block_theme() {
+	if ( ! did_action( 'setup_theme' ) ) {
+		_doing_it_wrong( __FUNCTION__, __( 'This function should not be called before themes are set up.' ), '6.8.0' );
```

Comment:
```suggestion
		_doing_it_wrong( __FUNCTION__, __( 'This function should not be called before theme directories are set up.' ), '6.8.0' );
```

Or 

`This function should not be called before register_theme_directory() is called.`